### ISSUE TITLE
[DateTimePicker] call onChange only after date AND time have been selected

### ISSUE THREAD

**Issue by ErnestBrandi** [mui/mui-x#4473]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Hi, I would like a way (new option ?) for `onChange` to be called only when the date AND the time have been selected by the user. Presently, the function is called after one of two is selected. For instance if we pick the date first, then the current time is set until we select the proper one.

**Comment by TheRealCuran** [comment]
This would really help me too.

If it helps with prioritising this ticket, my Order ID :credit_card: is: 47016

**Comment by flaviendelangle** [comment]
> Side note: I noticed, that the onChange doesn't seem to fire reliably for the time component. Basic reproducer code: https://codesandbox.io/s/green-cloud-p62c3k?file=/src/App.tsx (most common error case I am seeing is with just changing the time, but again: doesn't happen always, which makes this annoying to debug; I can reproduce it more reliably with my "big" application, the reproducer code seems sadly more stable)

Hi, could you create a standalone issue for this problem, to make sure we are tracking it correctly ?

____


Concerning the original topic of this issue
I fully agree with the need.

I don't know if we should add options to limit when `onChange` is called or if we should add information in the `onChange` to allow people ignoring it when the value is not complete.

**Comment by TheRealCuran** [comment]
Personally I would say: reduce the amount of events if possible, ie. your first option. Having too many events that caused a side effect was my reason for a lag in `DataGridPro`, so I'd always be in favour of having less events that can cause changes I might not want.

I vote for "add options to limit event firing".

(I'll file the separate report later/tomorrow, will have to retest this is still happing after I was out of office for a bit and there have been updates for `@mui/*`.)

**Comment by croraf** [comment]
If there is the OK button present, onChange should fire only when OK is clicked.

**Comment by LukasTy** [comment]
Closing this one as a duplicate of #6161.
